### PR TITLE
5829 - Updates to columns in CSV Download

### PR DESCRIPTION
**Pull request by PauloPortugal**
### What

A new column is required in the CSV file that is created for the Cantabular downloads. This should hold the category code for the row that the observation is being returned for.

* [5829 - Move "Count" column to the end of row](https://github.com/ONSdigital/dp-api-clients-go/commit/2b88db3115413ffa1aab5d2ea7230dcc590e2cb0)
Rather than the beginning of the row. The word `Count` should also be renamed to `Observation`

* [5829 - Add Category Code column for each Category Label](https://github.com/ONSdigital/dp-api-clients-go/commit/f33d77900bd966f4de81f76eb8aff0d7f67f9b39) 
A new column with the same Label string appended with `Code` should precede it's respective `Label` column.

**Resolves:** [5829](https://trello.com/c/N8L65t3B/5829-updates-to-columns-in-csv-download)

### How to review

1. Ensure tests pass
2. Sense check it
3. Check the story requirements

### Who can review

Any ONS developer

### CODE REVIEW

File: cantabular/table.go [hunk 1]
```diff
@@ -298,11 +298,12 @@ func decodeValues(ctx context.Context, dec jsonstream.Decoder, dims Dimensions,
...
 func createCSVHeader(dims Dimensions) []string {
-	header := make([]string, len(dims)+1)
+	l := len(dims)
+	header := make([]string, l+1)
 	for i, dim := range dims {
-		header[i+1] = dim.Variable.Label
+		header[i] = dim.Variable.Label
 	}
-	header[0] = "count"
+	header[l] = "Observation"
```
Comment (thomasridd) [inline cantabular/table.go:310]: Query: Should this be "Observation" or "Observations"
Ticket says "Observation" but I'd just like it checked

File: cantabular/table.go [hunk 2]
```diff
@@ -298,11 +298,16 @@ func decodeValues(ctx context.Context, dec jsonstream.Decoder, dims Dimensions,
 // createCSVHeader creates an array of strings corresponding to a csv header
 // where each column contains the value of the corresponding dimension, with the last column being the 'count'
 func createCSVHeader(dims Dimensions) []string {
-	header := make([]string, len(dims)+1)
-	for i, dim := range dims {
-		header[i+1] = dim.Variable.Label
+	i := 0
+	l := len(dims) * 2
+	header := make([]string, l+1)
+	for _, dim := range dims {
```
Comment (ChrysmOre) [inline cantabular/table.go:304]: Could you initialise `i` here in the loop declaration and remove the second incrementation?

File: cantabular/table.go (hunk 1 above)
Comment (thomasridd) [reply cantabular/table.go:310]: All good - ignore me

File: cantabular/table.go (hunk 2 above)
Comment (PauloPortugal) [reply cantabular/table.go:304]: I am not sure this would be a better option

Comment (PauloPortugal) [reply cantabular/table.go:304]: I would still stick with my initial approach, as it is easier to read the code.

Comment (j-s-rawat) [reply cantabular/table.go:304]: I am approving this as either approach are acceptable and won't effect the outcome

**Review by j-s-rawat — APPROVED** [review]
LGTM

**Pull request merged**